cmd/collector/app/collector.go

```diff
@@ -143,29 +144,27 @@ func (c *Collector) Close() error {
 	// HTTP server
 	if c.hServer != nil {
 		timeout, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-		err := c.hServer.Shutdown(timeout)
-		if err != nil {
-			c.logger.Error("failed to stop the main HTTP server", zap.Error(err))
+		if err := c.hServer.Shutdown(timeout); err != nil {
+			return fmt.Errorf("failed to stop the main HTTP server %w", err)
```

Comment:
I think logging here is more appropriate, because there is more work left for this function to do, and I am not certain the caller actually logs the returned error (although it should).

Reply:
You' re right, it's more safe.